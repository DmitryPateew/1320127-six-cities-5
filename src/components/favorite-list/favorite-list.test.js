import React from "react";
import renderer from "react-test-renderer";
import FavoriteList from "./favorite-list";
import {expect, it} from "@jest/globals";

const favoriteTest = [{
  id: 1,
  city: `Hamburg`,
  coords: [52.3909553943508, 4.85309666406198],
  favorite: true,
  photos: [`img/apartment-02.jpg`, `img/apartment-01.jpg`],
  title: `Beautiful & luxurious studio at great location`,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
  premium: false,
  typeOfHousing: `room`,
  rating: 1,
  countBedrooms: 3,
  maxAdults: 3,
  pricePerNight: 150,
  facilities: [`Wi-Fi`, `Heating`, `Kitchen`],
}, {
  id: 2,
  city: `Hamburg`,
  coords: [52.3909553943508, 4.85309666406198],
  favorite: true,
  photos: [`img/apartment-02.jpg`, `img/apartment-01.jpg`],
  title: `Beautiful & luxurious studio at great location`,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
  premium: false,
  typeOfHousing: `room`,
  rating: 1,
  countBedrooms: 3,
  maxAdults: 3,
  pricePerNight: 150,
  facilities: [`Wi-Fi`, `Heating`, `Kitchen`]
}];


it(`Should FavoriteList render correctly`, () => {
  const favoriteList = renderer.create(<FavoriteList
    offers={favoriteTest}
  />)
    .toJSON();

  expect(favoriteList).toMatchSnapshot();
});
