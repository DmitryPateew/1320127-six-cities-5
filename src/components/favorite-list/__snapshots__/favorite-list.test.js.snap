// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoriteList render correctly 1`] = `
<div
  className="favorites__places"
>
  <withRouter(FavoriteItem)
    key="0"
    offer={
      Object {
        "city": "Hamburg",
        "coords": Array [
          52.3909553943508,
          4.85309666406198,
        ],
        "countBedrooms": 3,
        "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
        ],
        "favorite": true,
        "id": 1,
        "maxAdults": 3,
        "photos": Array [
          "img/apartment-02.jpg",
          "img/apartment-01.jpg",
        ],
        "premium": false,
        "pricePerNight": 150,
        "rating": 1,
        "title": "Beautiful & luxurious studio at great location",
        "typeOfHousing": "room",
      }
    }
  />
  <withRouter(FavoriteItem)
    key="1"
    offer={
      Object {
        "city": "Hamburg",
        "coords": Array [
          52.3909553943508,
          4.85309666406198,
        ],
        "countBedrooms": 3,
        "description": "A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.",
        "facilities": Array [
          "Wi-Fi",
          "Heating",
          "Kitchen",
        ],
        "favorite": true,
        "id": 2,
        "maxAdults": 3,
        "photos": Array [
          "img/apartment-02.jpg",
          "img/apartment-01.jpg",
        ],
        "premium": false,
        "pricePerNight": 150,
        "rating": 1,
        "title": "Beautiful & luxurious studio at great location",
        "typeOfHousing": "room",
      }
    }
  />
</div>
`;
